Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -662,15 +662,19 @@ class FastStringifier {
     String firstGetterSetterPropertyName() const;
     void recordFastPropertyEnumerationFailure(JSObject&);
     bool haveFailure() const;
-    unsigned remainingCapacity() const;
+    bool hasRemainingCapacity(unsigned size = 1);
+    bool hasRemainingCapacitySlow(unsigned size);
     bool mayHaveToJSON(JSObject&) const;
 
     static void logOutcome(ASCIILiteral);
     static void logOutcome(String&&);
 
+    static unsigned useableBufferSize(size_t availableBufferSize);
+
     JSGlobalObject& m_globalObject;
     VM& m_vm;
-    unsigned m_length { 0 };
+    unsigned m_length { 0 }; // length of content already filled into m_buffer.
+    unsigned m_useableEnd { 0 };
```

Comment:
Ditto.

Typically we call this _capacity_. Maybe that’s a better name for here.

Reply:
Yes, `m_capacity` is a better name.  Changed.